infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutContextData.kt

```diff
@@ -34,6 +34,30 @@ interface LayoutContextAware<T> {
     val isDesign: Boolean get() = false
 }
 
+sealed class ContextIdentity<T> {
+    abstract val rowId: LayoutRowId<T>?
+}
+
+data class UnsavedIdentity<T>(
+    val sourceRowVersion: LayoutRowVersion<T>?,
```

Comment:
Tallentamaton olio ennen inserttiä: ei siis vielä row id:tä eikä versiota. Tilapäinen string-id generoidaan nyt täällä jotta oliolla on aina joku id (kuten ennenkin, aiemmin generoitiin contextdatassa).

Otin tänne mukaan tuon optionaalisen sourceRowVersion: se on alkuperäisen olion versio siinä tapauksessa että tämä olio on tehty kopioimalla joku olemassaoleva rivi (esim draftin luonti officialista). Suunnitelma muuttui toteuttaessa ja sillä ei tehdä tällä hetkellä mitään, mutta en poistanut koska se tuntui luonnolliselta lisältä debuggausta varten. Vähän ehkä suspect koska tarpeeton mutta kertoo mistä data tuli kun esim debuggaa jotain mergessä pieleen menevää juttua jne. :/

Reply:
Mua ei sinällään haittaa että jätetään kenttiä debug-tarkoituksiin (not the cleanest, mutta välillä tosiaan sillekin on paikkansa.) Nimeäisin tosin sen silloin jotenkin todella selkeän tökerösti (esim. `sourceRowVersion__FOR_DEBUG` tms.) Tämä sen takia, että koska tuo kenttä on a) nullable ja b) käyttämätön, se voi sisältää oikeastaan mitä vaan roskaa. Se voi kopsahtaa devaajaa nilkkaan jos tulevaisuudessa joku koittaakin käyttää sitä johonkin oikeaan juttuun.

Reply:
Ja huomiona luokan nimeen, kun alempana on käytetty termiä `Stored`, niin voisiko tämä olla `Unstored`?